sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CollationTypeCoercion.scala

```diff
@@ -223,151 +277,195 @@ object CollationTypeCoercion {
     val collationContextWinner = expressions.foldLeft(findCollationContext(expressions.head)) {
       case (Some(left), right) =>
         findCollationContext(right).flatMap { ctx =>
-          collationPrecedenceWinner(left, ctx)
+          mergeWinner(left, ctx)
         }
       case (None, _) => return None
```

Comment:
Not related to the PR, but `return` is not necessary, it seems.

Reply:
nice catch, removed the return